src/chocolatey.resources/helpers/functions/Uninstall-ChocolateyPath.ps1

```diff
@@ -0,0 +1,174 @@
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+function Uninstall-ChocolateyPath {
+<#
+.SYNOPSIS
+**NOTE:** Administrative Access Required when `-PathType 'Machine'.`
+
+This puts a directory to the PATH environment variable of the
+requested scope (Machine or User).
+
+.DESCRIPTION
+Removes path from target path scope.  Removes multiple occurances (if they exist)
+and all occurances with or without a trailing slash.
+
+.NOTES
+This command will assert UAC/Admin privileges on the machine if
+`-PathType 'Machine'`.
+
+This is used when the application/tool is not being linked by Chocolatey
+(not in the lib folder).
+
+.INPUTS
+None
+
+.OUTPUTS
+None
+
+.PARAMETER PathToUninstall
+The full path to a location to remove from the PATH.
+
+.PARAMETER PathType
+Which PATH to remove it from. If specifying `Machine`, this requires admin
+privileges to run correctly.
+
+.PARAMETER IgnoredArguments
+Allows splatting with arguments that do not apply. Do not use directly.
+
+.EXAMPLE
+Uninstall-ChocolateyPath -PathToUninstall "$($env:SystemDrive)\tools\gittfs"
+
+.EXAMPLE
+Uninstall-ChocolateyPath "$($env:SystemDrive)\Program Files\MySQL\MySQL Server 5.5\bin" -PathType 'Machine'
+
+.LINK
+Install-ChocolateyPath
+
+.LINK
+Install-ChocolateyEnvironmentVariable
+
+.LINK
+Get-EnvironmentVariable
+
+.LINK
+Set-EnvironmentVariable
+
+.LINK
+Get-ToolsLocation
+#>
+param(
+  [parameter(Mandatory=$true, Position=0)][alias("Path")][string] $PathToUninstall,
+  [parameter(Mandatory=$false, Position=1)][ValidateSet('User','Machine','All')][alias("Scope")][String] $pathType = 'User',
+  [parameter(Mandatory=$false)][switch] $RecursiveCall,
+  [parameter(ValueFromRemainingArguments = $true)][Object[]] $ignoredArguments
+)
+  Write-Debug "Running 'Uninstall-ChocolateyPath' with PathToUninstall:`'$PathToUninstall`'";
+  If (!$RecursiveCall -AND ($pathType -ine 'All')) {Write-Host "Only evaluating and updating path scope `"$pathType`", path will not be assessed nor removed for other scope, so path may exist in other scope as well."}
+  $originalPathToUninstall = $PathToUninstall
+  #First half on handling trailing slash properly - remove it from requested path:
+  $PathToUninstall = $PathToUninstall.trimend('\')
+  #array drops blanks (one of which is always created by final semi-colon)
+  $actualPathArrayUser = (Get-EnvironmentVariable -Name 'Path' -Scope 'user' -PreserveVariables).split(';',[System.StringSplitOptions]::RemoveEmptyEntries)
+  $actualPathArrayMachine = (Get-EnvironmentVariable -Name 'Path' -Scope 'machine' -PreserveVariables).split(';',[System.StringSplitOptions]::RemoveEmptyEntries)
```

Comment:
We're going to have to do something a little smarter since apparently [semicolons can be escaped using quotes](https://superuser.com/a/594177). Too bad there isn't a Windows API for this.

Reassembling the path will have to follow the same procedure in reverse.

Reply:
Confirmed using Windows' environment variable list editor: a `tl;dr` entry causes `;"tl;dr";` to be added to the path.

Reply:
Also, the path needs to be recognized and removed even if it is quoted in the PATH variable. I assume the input path would never be quoted.

Reply:
So luckily PATH doesn't have `"` and it doesn't have escaped `;` - if you make it have `"`, it doesn't like you anymore IIRC.

Reply:
Am I understanding that a semi-colon is allowed in a folder name???!!!

Reply:
I just died a little inside. 👎